snewpdag/plugins/gen/GenPoint.py

```diff
@@ -42,11 +48,14 @@ def __init__(self, detector_location, ra, dec, time, **kwargs):
     self.ra = np.radians(ra)
     self.dec = np.radians(dec)
     self.time = Time(time)
+    self.time_unix = self.time.to_value('unix', 'long') # float, unix epoch
+    self.epoch_base = kwargs.pop('epoch_base', 0.0)
+
+    if not isinstance(self.epoch_base, (numbers.Number, str, list, tuple)):
+      logging.error('GenPoint.__init__: unrecognized epoch_base {}. Set to 0.'.format(self.epoch_base))
+      self.epoch_base = 0.0
+
```

Comment:
which was your final choice of the reference epoch after the comment by Rafael? is that the epoch_base here?

Reply:
Yes, this is epoch_base.  It actually doesn't matter too much; the main requirement is that it's consistent within one invocation of snewpdag.  So we can fix a time, or let the SNEWS implementation folks set it for us.  There's a small but nonzero risk associated with an auto-rollover, though - or shall we say complication, because one would have to add logic just in case something happens around the rollover.